sympy/solvers/ode.py

```diff
@@ -4003,6 +4002,43 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+
+def _check_substitution_type(eq, func):
```

Comment:
This function should be called _order_reducing_substitution_match

Reply:
In what way?Like I added "ode" in front of the other function or I should change the name because i is very well being detected and even working.

Reply:
Ok got that and done that. Again this "test_ode.py" file is full of assert which I have never used. Googling it tells that as the name suggests it checks for assertions but then what is that "_order_reducing_substitution_match" made for?

Reply:
And also I tried looking for spaces.I couldn't find them.What to do?